internal/sgbouncewizard/sgbouncewizard.go

```diff
@@ -91,116 +112,172 @@ func (srv *Server) CheckUser(w http.ResponseWriter, r *http.Request) {
 		result.FirstName = user[0].FirstName
 		result.LastName = user[0].LastName
 		result.Role = user[0].Role
+	} else {
+		passError :=  errors.New("verifyPassword Failed")
+		http.Error(w, passError.Error(), http.StatusBadRequest)
+		return
 	}
 
 	data, err := json.Marshal(result)
 
 	if err != nil {
+		log.Println(err)
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
 
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
 	w.Write(data)
 }
 
 // ListRules - wrapper to grab all rules
-func  (srv *Server) ListRules(w http.ResponseWriter, r *http.Request) {
-	rules, err := srv.DBClient.ListRules()
-	// fmt.Println(rules)
+func  (srv *Server) getAllRulesRoute(w http.ResponseWriter, r *http.Request) {
+	rules, err := srv.DBClient.GetAllRules()
+
+	if err != nil {
+		log.Println(err)
+		http.Error(w, err.Error(), http.StatusNotFound)
+		return
+	}
 
 	data, err := json.Marshal(&rules)
 
 	if err != nil {
+		log.Println(err)
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
 
 	w.Header().Set("Access-Control-Allow-Origin", "*")
-	w.WriteHeader(200)
 	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
 	w.Write(data)
 }
 
-func (srv *Server) getRule(w http.ResponseWriter, r *http.Request) {
-	w.Header().Set("Content-Type", "application/json")
-	rule := r.Context().Value("rule").(*models.BounceRule)
+func (srv *Server) getRuleRoute(w http.ResponseWriter, r *http.Request) {
+	rule , ok := r.Context().Value("rule").(*models.BounceRule)
+
+	if !ok {
+		log.Println(ok)
```

Comment:
This `log.print ` line might be hard to decipher the meaning in the logs. I think all you will see is 

```
Dec 27 03:00:35 bouncecm: false
Dec 27 03:00:36 bouncecm: false
```
Consider adding more context to the logged statement?

Reply:
changed all `log.println(ok)` to  `log.Println("ContextValue of rule in GetRuleRoute: " + strconv.FormatBool(ok))`